presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -214,8 +214,6 @@ public PlanNode visitProject(ProjectNode node, RewriteContext<Expression> contex
         @Override
         public PlanNode visitGroupId(GroupIdNode node, RewriteContext<Expression> context)
         {
-            checkState(!SymbolsExtractor.extractUnique(context.get()).contains(node.getGroupIdSymbol()), "groupId symbol cannot be referenced in predicate");
```

Comment:
this and below change in this file should not belong to separate commit? IMO one thing is to inline grouping function in analyzer and the other thing is to add support for groupid be present in predicate. What do you think?

Also I think this checkState could be removed separately, as groupId cannot be used in common grouping symbols so it will be used only in wrapping filter node.

Reply:
Yeah, I can pull it out into a separate commit